Backend/app/oauth_handler.py

```
@@ -1,14 +1,83 @@
+        return state, None
+
+    encoded = state[len(STATE_PREFIX):]
+    padding = len(encoded) % 4
+    if padding:
+        encoded += "=" * (4 - padding)
+
+    try:
+        decoded = base64.urlsafe_b64decode(encoded.encode("utf-8"))
+        payload = json.loads(decoded)
+        return payload.get("state", ""), payload.get("client_redirect_uri")
+    except Exception:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)